File: server/api/config/getDeviceId.go
package config

import (
	"github.com/gin-gonic/gin"
	"github.com/google/uuid"
)

func GetDeviceId(c *gin.Context) {
	deviceId, err := c.Cookie("deviceId")
	if err != nil {
		deviceId = uuid.NewString()
		c.SetCookie("deviceId", deviceId, 100000000, "/", "localhost", false, true)
	}
	c.String(200, deviceId)
}

File: src/sdk/server/getDeviceId.ts
import { v4 as uuid } from "uuid";

function getDeviceId() {
  let deviceId = localStorage.getItem("deviceId");
  if (!deviceId) {
    deviceId = uuid();
    localStorage.setItem("deviceId", deviceId);
  }
  return deviceId;
}


export default getDeviceId